src/comms.cpp

```diff
@@ -693,10 +716,10 @@ void clover_send_recv_message_left(
 
 	int left_task = globals.chunk.chunk_neighbours[chunk_left] - 1;
 
-	MPI_Isend(globals.chunk.left_snd.actual(), total_size, MPI_DOUBLE, left_task, tag_send,
+	MPI_Isend(globals.chunk.left_snd.data, total_size, MPI_DOUBLE, left_task, tag_send,
 	          MPI_COMM_WORLD, &req_send);
 
-	MPI_Irecv(globals.chunk.left_rcv.actual(), total_size, MPI_DOUBLE, left_task, tag_recv,
+	MPI_Irecv(globals.chunk.left_rcv.data, total_size, MPI_DOUBLE, left_task, tag_recv,
```

Comment:
Here are the arrays that must live on the host. I'm not sure I've spotted where we `update` them on the device again.
Similarly, we need to `update from` on the host before the `MPI_Isend`

Reply:
Fixed in https://github.com/UoB-HPC/cloverleaf_openmp_target/pull/2/commits/37d197ade0f4ff330fd969c20502a7a86100139f